src/Agent.Worker/DiagnosticLogManager.cs

```diff
@@ -148,6 +148,18 @@ public sealed class DiagnosticLogManager : AgentService, IDiagnosticLogManager
                 executionContext.Debug("Dumping cloud-init logs is ended.");
             }
 
+            // Copy event logs for windows machines
+            if (PlatformUtil.RunningOnWindows)
+            {
+                executionContext.Debug("Dumping event viewer logs for current job.");
+
+                string eventLogsFile = $"{HostContext.GetDirectory(WellKnownDirectory.Diag)}/EventViewer-{ jobStartTimeUtc.ToString("yyyyMMdd-HHmmss") }.log";
+                await DumpCurrentJobEventLogs(executionContext, eventLogsFile, jobStartTimeUtc);
+
+                string destination = Path.Combine(supportFilesFolder, Path.GetFileName(eventLogsFile));
+                File.Copy(eventLogsFile, destination);
```

Comment:
What happens if file copying fails? Could it make sense to wrap all this logic with try..catch?

Reply:
If file copy fails, then the entire logs archive won't be uploaded to ADO. Wrapped it all in try-catch, thanks!